packages/synapse-react-client/src/synapse-queries/chat/useChat.ts

```diff
@@ -37,46 +30,60 @@ export function useCreateAgentSession(
     mutationFn: (request: CreateAgentSessionRequest) =>
       SynapseClient.createAgentSession(request, accessToken),
     onSuccess: async (newAgentSession, variables, ctx) => {
-      const queryKey = keyFactory.getAgentSessionQueryKey(
-        newAgentSession.sessionId,
-      )
-      queryClient.invalidateQueries({
-        queryKey: queryKey,
-      })
-
       if (options?.onSuccess) {
         await options.onSuccess(newAgentSession, variables, ctx)
       }
     },
   })
 }
 
-export function useGetAgentChatWithAsyncStatus<
-  TData = AsynchronousJobStatus<AgentChatRequest, AgentChatResponse>,
->(
-  request: AgentChatRequest,
-  options?: Partial<
-    UseQueryOptions<
-      AsynchronousJobStatus<AgentChatRequest, AgentChatResponse>,
-      SynapseClientError,
-      TData
-    >
+export function useSendChatMessageToAgent(
+  options?: UseMutationOptions<
+    AgentChatResponse,
+    SynapseClientError,
+    AgentChatRequest
   >,
   setCurrentAsyncStatus?: (
     status: AsynchronousJobStatus<AgentChatRequest, AgentChatResponse>,
   ) => void,
 ) {
-  const { accessToken, keyFactory } = useSynapseContext()
-
-  return useQuery({
-    ...options,
-    queryKey: keyFactory.getAgentChatWithAsyncStatusQueryKey(request),
-
-    queryFn: () =>
-      SynapseClient.getAgentChatAsyncJobResults(
+  const { accessToken } = useSynapseContext()
+  return useMutation<
+    AsynchronousJobStatus<AgentChatRequest, AgentChatResponse>,
+    SynapseClientError,
+    AgentChatRequest
+  >({
+    mutationFn: (request: AgentChatRequest) => {
+      return SynapseClient.getAgentChatAsyncJobResults(
         request,
         accessToken,
         setCurrentAsyncStatus,
-      ),
+      )
+    },
+    onSuccess: (data, variables, ctx) => {
+      if (options?.onSuccess && data.responseBody) {
```

Comment:
May optimistically update the history cache (once it uses useQuery). It might get weird with pagination, but hopefully that isn't too complicated to solve.